build/targets/KoreBuild.Common.targets

```diff
@@ -10,7 +10,8 @@ extending the *DependsOn property
 -->
   <PropertyGroup>
     <PrepareDependsOn></PrepareDependsOn>
-    <CompileDependsOn>Prepare;Restore</CompileDependsOn>
+    <CompileDependsOn>Prepare</CompileDependsOn>
+    <CompileDependsOn Condition="'$(NoRestore)'=='false' Or '$(NoRestore)'==''">$(CompileDependsOn);Restore</CompileDependsOn>
```

Comment:
Also add the spaces around the conditional operator here and in the new condition below.